js-core/homeworks/phoneApp/src/main.js

```diff
@@ -0,0 +1,111 @@
+function PhoneApp() {
+    this.storage = [];
+}
+
+PhoneApp.prototype.checkNumber = function(numberToCheck) {
+    const toStringThePredictNumber = numberToCheck.toString();
+    const toArrayTheString = toStringThePredictNumber.split('');
+    let flag;
+    
+    toArrayTheString.forEach(elem => {
+        const toNumberElem = Number(elem);
+        if(isNaN(toNumberElem)) {
+            flag = false;
+        }
+    });
+    
+    if(flag === false) {
+        return flag;
+    } else {
+        return true;
+    }
+}
+
+PhoneApp.prototype.sortNumber = function(numberToSort) {
+    if(this.checkNumber(numberToSort)) {
+        const toStringTheNumber = numberToSort.toString();
+        const firstThreeNumbers = toStringTheNumber.slice(0, 3);
+        const bracketThisFirstThreeNumbers = `(${firstThreeNumbers}) `;
+        
+        const secondTwoNumbers = toStringTheNumber.slice(3, 5);
+        const bracketThisSecondTwoNumbers = `${secondTwoNumbers}-`;
+        
+        const thirdTwoNumbers = toStringTheNumber.slice(5, 7);
+        const bracketThisThirdTwoNumbers = `${thirdTwoNumbers}-`;
+        
+        let lastNumbers = '';
+        for(let i = 7; i < toStringTheNumber.length; i++) {
+            const elem = toStringTheNumber[i];
+            lastNumbers += elem;
+        }
+        
+        const finalConstraction = bracketThisFirstThreeNumbers + bracketThisSecondTwoNumbers + bracketThisThirdTwoNumbers + lastNumbers;
+        
+        return finalConstraction;
+    } else {
+        return console.log('EROR you had typed incorect number')
+    }
+}    
+
+PhoneApp.prototype.add = function(name, number, surname = null, city = null, company = null) {
```

Comment:
there a lot of arguments, it's better to pass 1 object(with options) instead of 5 arguments